presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java

```diff
@@ -66,7 +68,9 @@
     private long hashCollisions;
     private double expectedHashCollisions;
 
-    public BigintGroupByHash(int hashChannel, boolean outputRawHash, int expectedSize)
+    private final Function<Long, Boolean> updateReservedMemory;
```

Comment:
Introduce an interface for this. You can continue using lambda with a custom interface.